packages/api/models/granules.js

```diff
@@ -305,41 +322,71 @@ class Granule extends Manager {
   /**
    * Store a granule record in DynamoDB.
    *
-   * @param {Object} granuleRecord - A granule record.
+   * @param {Object} granuleRecord           - A granule record.
+   * @param {boolean} writeConstraints       - boolean toggle restricting if write conditionals
+   *                                           should be compared to existing record to determine
+   *                                           write eligibility
    * @returns {Promise<Object|undefined>}
    */
-  async _storeGranuleRecord(granuleRecord) {
-    const mutableFieldNames = this._getMutableFieldNames(granuleRecord);
+  async _storeGranuleRecord(granuleRecord, writeConstraints = true) {
+    const mutableFieldNames = this._getMutableFieldNames(granuleRecord, writeConstraints);
     const updateParams = this._buildDocClientUpdateParams({
       item: granuleRecord,
       itemKey: { granuleId: granuleRecord.granuleId },
       mutableFieldNames,
     });
 
-    // createdAt comes from cumulus_meta.workflow_start_time
-    // records should *not* be updating from createdAt times that are *older* start
-    // times than the existing record, whatever the status
-    updateParams.ConditionExpression = '(attribute_not_exists(createdAt) or :createdAt >= #createdAt)';
-
-    // Only allow "running" granule to replace completed/failed
-    // granule if the execution has changed for granules with executions.
-    // Allow running granule to replace queued granule
-    if (granuleRecord.status === 'running' && granuleRecord.execution !== undefined) {
-      updateParams.ExpressionAttributeValues[':queued'] = 'queued';
-      updateParams.ConditionExpression += ' and (#status = :queued or #execution <> :execution)';
-    }
+    if (writeConstraints === true) {
+      if (granuleRecord.createdAt) {
```

Comment:
Do we need this in JS? It seems like the dynamo conditional might handle the check for `createdAt`?

`'(attribute_not_exists(createdAt) or :createdAt >= #createdAt)'`

Reply:
We do, because 

```
    Object.entries(item).forEach(([fieldName, value]) => {
      if (itemKeyFieldNames.includes(fieldName)) return;
      if (value === undefined) return;

      ExpressionAttributeNames[`#${fieldName}`] = fieldName;
      ExpressionAttributeValues[`:${fieldName}`] = value;

      if (mutableFieldNames.includes(fieldName)) {
        setUpdateExpressions.push(`#${fieldName} = :${fieldName}`);
      } else {
        setUpdateExpressions.push(`#${fieldName} = if_not_exists(#${fieldName}, :${fieldName})`);
      }
    });
```

is built into `_buildDocClientUpdateParams` and for `:createdAt`/`#createdAt` to have meaning you have to have those values set. 

We could, in theory, find another way to organize the logic such that it was more directly coupled, but I didn't find a convincing way that didn't read at least as complicated.    Given the projected longevity of this model, I'm inclined to *not* attempt a refactor of that nature, but very open to other opinions/solutions.